operations/observability/Observability.md

```diff
@@ -0,0 +1,314 @@
+```
+
+### Grok-exporter deployment, service, and service-monitor
+
+The deployment for the grok-exporter is for all the master nodes by setting the node affinity as `ncn-m001`, `ncn-m002`, and `ncn-m003` for NCN deployment of grok-exporter.
+The service for the grok-exporter is accessible at port `9144`. Service-monitor implementation is for Prometheus to access the metrics that are created by the grok-exporter.
+
+### Configuration file for the grok-exporter
+
+The configuration file for the grok-exporter parses the Goss test log file and creates metrics from them.
+grok-exporter version 3 configuration is used for this task and to match log expressions with regular expressions.
+
+The following is an example configuration passed to the grok-exporter to get metrics:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: example_name
+  namespace: example_namespace
+data:
+  config.yml: |-
+    global:
+      config_version: 3
+    input:
+      type: file
+      paths: 
+      - /logs/goss_tests/*
+      fail_on_missing_logfile: false
+    grok_patterns:
+      - 'REGEX [regular].*expression'
+      - 'EXAMPLE [aA-zZ-].*'
+    metrics:
+      - type: gauge
+        name: example_metric
+        help: EXAMPLE METRIC HELP
+        match: '{"log_file": "%{REGEX:logfile}", "TEST": "%{EXAMPLE:ex}"}'
+        value: '{{`{{.logfile}}`}}'
+        labels:
+          example_test: '{{`{{.ex}}`}}'
+    server:
+     port: 9144
+```
+
+### Prometheus metrics and Grafana dashboard
+
+After the preceding steps the `goss_tests` metrics are seen in Prometheus when Goss tests are run. Using these metrics, Grafana dashboards are created to shows the Goss tests details visually.
+
+Goss test dashboard features:
+
+- The dashboard has dropdown for product, suite, and tests.
+- By default, all products, suites, and tests are selected for overall Goss tests result.
+- The overall product result, total number of the products, products passed, products failed, and its execution time is seen.
+- Suite results and test results are seen from the dashboard.
+- Node-wise test result are seen with failed nodes, passed nodes, and the suite/test description.
+
+![Goss test dashboard](../../img/operations/GossTestsDashboard.png "Goss Test Dashboard")
+
+## Error dashboard
+
+Error dashboards provide key glance at-a-insights about what is broken and needs attention. This works for both internal HPE environments and clusters as well as is a containerized shippable solution that customers can use as well.
```

Comment:
Isn't it assumed by default that our metrics work both internally and on customer deployments, unless we say otherwise? Or am I missing a subtlety here?
```suggestion
Error dashboards provide key glance at-a-insights about what is broken and needs attention.
```

Reply:
Changed.